src/frequenz/client/electricity_trading/_client.py

```diff
@@ -116,15 +121,28 @@ def __new__(
 
         return cls._instances[key]
 
-    def __init__(
-        self, server_url: str, connect: bool = True, auth_key: str | None = None
+    def __init__(  # pylint: disable=too-many-arguments, too-many-positional-arguments
+        self,
+        server_url: str,
+        connect: bool = True,
+        auth_key: str | None = None,
+        initial_timeout: int = 10,
+        max_timeout: int = 300,
+        timeout_increment: int = 20,
+        max_timeout_retries: int = 30,
+        reset_interval: int = 600,
     ) -> None:
         """Initialize the client.
 
         Args:
             server_url: The URL of the Electricity Trading service.
             connect: Whether to connect to the server immediately.
             auth_key: The API key for the authorization.
+            initial_timeout: Initial timeout duration for gRPC calls (in seconds).
+            max_timeout: Maximum timeout duration for gRPC calls (in seconds).
+            timeout_increment: Increment value for timeout on each retry (in seconds).
```

Comment:
I would suggest making this a timeout multiplier instead. Instead of adding a value, you would then multiply it with this factor. That way you can achieve exponential backing off easily.

Reply:
Oh okay I actually purposefully did an increment and not multiplier as I thought having an exponential increase would be too much here. But I can change it to multiplier for sure if it's better practice!